resources/js/components/SlideIn.vue

```diff
@@ -0,0 +1,83 @@
+<script>
+    import { usePointerSwipe } from '@vueuse/core'
+
+    export default {
+        props: {
+            direction: {
+                type: String,
+                default: 'up'
+            },
+            threshold : {
+                type: Number,
+                default: 0.5,
+            }
+        },
+
+        render() {
+            return this.$scopedSlots.default(this)
+        },
+
+        data() {
+            return {
+                target: null,
+                container: null,
+                left: '0',
+                top: '0',
+                isSwiping: false,
+                distanceX: 0,
+                distanceY: 0,
+                checked: false,
```

Comment:
Checked may be a bit too specific to our use case with a checkbox.
maybe something like `isOpen` is a better name here.